ElfPatcher.cs

```diff
@@ -22,16 +31,20 @@ class ElfPatcher {
 
 			// decrypt
 			if ( worker != null ) { worker.ReportProgress( 0, "Decrypting..." ); }
-			string elfPath = Path.Combine( Path.GetDirectoryName( ebootModPath ), Path.GetFileName( ebootPath ) + "-mod.ELF" );
+			string elfPath = Path.Combine( Path.GetDirectoryName( patcherPath ), Path.GetFileName( ebootPath ) + "-mod.ELF" );
 			int tries = 5;
 			while ( !( File.Exists( elfPath ) && Patcher.CalcMd5( elfPath ) == "a424aa775b707539dbff08cdb2e61ff5" ) ) {
 				if ( --tries < 0 ) {
 					throw new PatchingException( "Could not decrypt EBOOT. Confirm that EBOOT is correctly ripped and ebootMOD is working correctly." );
 				}
 
-				// this is super ugly but the only sensible way since calling unself directly searches the keys who-knows-where
 				try {
-					RunEbootModAndKill( ebootModPath, "\"" + ebootPath + "\"" );
+					if (linuxOrMac) {
+						RunElfPatcherAndKill( patcherPath, " " + ebootPath + " " + elfPath);
```

Comment:
`RunElfPatcherAndKill` is some real hacky code that runs ebootmod and then kills it mid-execution to get the decrypted eboot, you should probably call a regular program execution here instead.